### PR TITLE
feat(console): dashboard active users and curve

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add active users blocks and daily active user line chart.

## Linear Issue Reference

LOG-1698
LOG-1699

## Testing

截屏2022-06-08 下午1 59 59

### CODE REVIEW

**Comment by demonzoo** [comment]
image

Wow, why?

**Comment by wangsijie** [comment]
@demonzoo lock file fixed.

File: packages/console/src/pages/Dashboard/index.tsx [hunk 1]
```diff
@@ -23,35 +33,102 @@ const Dashboard = () => {
...
+                </div>
+              </div>
+            </div>
+            <div className={styles.curve}>
+              <ResponsiveContainer>
+                <AreaChart
+                  data={activeData.dauCurve.map((item) => ({
+                    ...item,
+                    // Remove "year" for a compact label.
+                    date: item.date.replace(/\d{4}-/, ''),
```
Comment (xiaoyijun) [inline packages/console/src/pages/Dashboard/index.tsx:85]: nit: Remenber we have `day.js` in our console.
```js
dayjs(item.date).format('MM-DD');
```

Comment (wangsijie) [reply packages/console/src/pages/Dashboard/index.tsx:85]: The `date` returned from API does not contain timezone info, so is not safe to parse it into a date.

**Review by xiaoyijun — APPROVED** [review]
LGTM

File: packages/console/src/pages/Dashboard/index.tsx [hunk 2]
```diff
@@ -23,35 +33,102 @@ const Dashboard = () => {
...
-              <div className={styles.number}>{newData.today.count}</div>
-              <div className={classNames(styles.delta, newData.today.delta < 0 && styles.down)}>
-                ({newData.today.delta > 0 && '+'}
-                {newData.today.delta}){newData.today.delta > 0 ? <ArrowUp /> : <ArrowDown />}
+        <>
+          <div className={styles.blocks}>
+            <Card className={styles.block}>
+              <div className={styles.title}>{t('dashboard.total_users')}</div>
+              <div className={styles.content}>
+                <div className={styles.number}>{totalData.totalUserCount}</div>
```
Comment (demonzoo) [inline packages/console/src/pages/Dashboard/index.tsx:41]: nit: Should we format the numbers with comma, like: 3,020

File: packages/console/src/pages/Dashboard/index.tsx [hunk 3]
```diff
@@ -23,35 +33,102 @@ const Dashboard = () => {
...
-                {newData.last7Days.delta})
-                {newData.last7Days.delta > 0 ? <ArrowUp /> : <ArrowDown />}
+            </Card>
+            <Card className={styles.block}>
+              <div className={styles.title}>{t('dashboard.new_users_today')}</div>
+              <div className={styles.content}>
+                <div className={styles.number}>{newData.today.count}</div>
+                <div className={classNames(styles.delta, newData.today.delta < 0 && styles.down)}>
+                  ({newData.today.delta > 0 && '+'}
+                  {newData.today.delta}){newData.today.delta > 0 ? <ArrowUp /> : <ArrowDown />}
```
Comment (demonzoo) [inline packages/console/src/pages/Dashboard/index.tsx:50]: `newData.today.delta` is used multiple times here. Should extract a const

File: packages/console/src/pages/Dashboard/index.tsx [hunk 4]
```diff
@@ -23,35 +33,102 @@ const Dashboard = () => {
...
+            <Card className={styles.block}>
+              <div className={styles.title}>{t('dashboard.new_users_7_days')}</div>
+              <div className={styles.content}>
+                <div className={styles.number}>{newData.last7Days.count}</div>
+                <div
+                  className={classNames(styles.delta, newData.last7Days.delta < 0 && styles.down)}
+                >
+                  ({newData.last7Days.delta > 0 && '+'}
+                  {newData.last7Days.delta})
+                  {newData.last7Days.delta > 0 ? <ArrowUp /> : <ArrowDown />}
```
Comment (demonzoo) [inline packages/console/src/pages/Dashboard/index.tsx:63]: same for `newData.lastUDays.delta`

Comment (demonzoo) [reply packages/console/src/pages/Dashboard/index.tsx:63]: Also, what if the delta === 0? Shouldn't show arrow down when delta is 0

File: packages/console/src/pages/Dashboard/index.tsx [hunk 5]
```diff
@@ -23,35 +33,102 @@ const Dashboard = () => {
...
+                  <div className={styles.number}>{activeData.wau.count}</div>
+                  <div
+                    className={classNames(styles.delta, activeData.wau.delta < 0 && styles.down)}
+                  >
+                    ({activeData.wau.delta > 0 && '+'}
+                    {activeData.wau.delta}){activeData.wau.delta > 0 ? <ArrowUp /> : <ArrowDown />}
+                  </div>
+                </div>
+              </Card>
+              <Card className={styles.block}>
```
Comment (demonzoo) [inline packages/console/src/pages/Dashboard/index.tsx:117]: These cards look similar, consider extracting it into a component

File: packages/console/src/pages/Dashboard/index.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/console/src/pages/Dashboard/index.tsx]: refactored

File: packages/console/src/pages/Dashboard/index.tsx (hunk 5 above)
Comment (wangsijie) [reply packages/console/src/pages/Dashboard/index.tsx]: refactored

**Comment by wangsijie** [comment]
截屏2022-06-09 下午1 46 00

File: packages/console/src/pages/Dashboard/index.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/Dashboard/index.tsx]: updated

**Review by simeng-li — APPROVED** [review]

File: packages/console/src/pages/Dashboard/index.tsx [hunk 6]
```diff
@@ -23,35 +32,68 @@ const Dashboard = () => {
...
-            <div className={styles.title}>{t('dashboard.total_users')}</div>
-            <div className={styles.content}>
-              <div className={styles.number}>{totalData.totalUserCount}</div>
+        <>
+          <div className={styles.blocks}>
+            <Block title="dashboard.total_users" count={totalData.totalUserCount} />
+            <Block
+              title="dashboard.new_users_today"
+              count={newData.today.count}
+              delta={newData.today.delta}
```
Comment (demonzoo) [inline packages/console/src/pages/Dashboard/index.tsx:41]: nit: Block can accept a `data` prop, which consists of both `count` and `delta`

**Review by demonzoo — APPROVED** [review]

**Pull request merged**